### PR TITLE
Add pain and fatigue to the morale window

### PR DESCRIPTION
**Pull request by AlexMooney**
#### Summary

```SUMMARY: Interface "Adds pain and fatigue penalties on the morale screen, when applicable"```

#### Purpose of change

It can be confusing to see "Total morale +19" and yet "Focus trends toward 62".

#### Describe the solution

Add the pain and fatigue effects to the display.  Don't take up the lines when one or both are not present.

| normal | pain | both |
| --- | --- | --- |
| image | image | image |

#### Describe alternatives you've considered

I considered leaving the pain and fatigue modifiers always present on the display, but thought that'd be cluttered.  I also considered hacking the penalties to appear with the other negative morale sources, but that seemed like it would get messy.

#### Testing

Basically the scenarios in the screenshots.

### CODE REVIEW

**Comment by Siersha** [comment]
I'm all for the change, however, calc_focus_equilibrium() was changed to include a bool parameter. Other sections in code need to include this as well.
image

**Comment by AlexMooney** [comment]
> I'm all for the change, however, calc_focus_equilibrium() was changed to include a bool parameter. Other sections in code need to include this as well.

No, I added `int calc_focus_equilibrium( bool ignore_pain = false ) const;` in avatar.h so as not to change the signature.  If you don't pass `true`, then you get the current behavior.  The compiler was not happy when I tried to include the default arg in the cpp file, so it's not obvious enough IMO.

**Comment by Siersha** [comment]
> > I'm all for the change, however, calc_focus_equilibrium() was changed to include a bool parameter. Other sections in code need to include this as well.
> 
> No, I added `int calc_focus_equilibrium( bool ignore_pain = false ) const;` in avatar.h so as not to change the signature. If you don't pass `true`, then you get the current behavior. The compiler was not happy when I tried to include the default arg in the cpp file, so it's not obvious enough IMO.

Did you run astyle with the changed files? travis build tends to fail for that

**Comment by AlexMooney** [comment]
> Did you run astyle with the changed files? travis build tends to fail for that

Yeah, I build the game with the Makefile, so it doesn't compile with astyle regressions.  The Travis CI shows astyle is happy [here](https://travis-ci.org/CleverRaven/Cataclysm-DDA/jobs/597324231#L989).

File: src/morale.cpp [hunk 1]
```diff
@@ -451,14 +451,16 @@ void player_morale::decay( const time_duration ticks )
...
 
-void player_morale::display( int focus_eq )
+void player_morale::display( int focus_eq, int pain_penalty, int fatigue_penalty )
 {
     /*calculates the percent contributions of the morale points,
      * must be done before anything else in this method
      */
     calculate_percentage();
 
     const char *morale_gain_caption = _( "Total morale:" );
```
Comment (ZhilkinSerg) [inline src/morale.cpp]: Huh, I wonder why it was not even a `std::string`...

Comment (AlexMooney) [reply src/morale.cpp]: It seems to be translated:

```
lang/po/zh_CN.po
171779-#, c-format
171780:msgid "Total morale: %d"
171781-msgstr "总心情值：%d"
```

https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/TRANSLATING.md#translation-functions
